### DISCUSSION TITLE
I have a trouble in verifying the smart contract on XDC test network (Apothem) using hardhat.

### DISCUSSION THREAD

**Discussion by workwithhim** [community/community#64454] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I deployed a NFT smart contract on Apothem Network using hardhat.
But when I am going to verify this smart contract, I have the following issue.
 ``npx hardhat verify --network apothem --constructor-args scripts/NFT_var.ts 0x7DE1DB267626360C0172a466FEbb8267B4570980``

``SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at consumeEnd (F:\...\node_modules\undici\lib\api\readable.js:253:20)
    at BodyReadable.<anonymous> (F:\...\node_modules\undici\lib\api\readable.js:235:7)
    at BodyReadable.emit (node:events:513:28)
    at BodyReadable.emit (node:domain:489:12)
    at BodyReadable.emit (F:\...\node_modules\undici\lib\api\readable.js:66:18)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)``

I am using hardhat compiler 0.8.12;
How can I fix this error?

**Comment by jareehmonda** [comment]
It seems [like](https://winchinsider.com/how-to-put-winch-in-free-spool/) you're facing a JSON parsing issue while trying to verify your NFT smart contract on the Apothem Network using Hardhat. This might be due to incorrect formatting or content in the verification arguments. To [fix this](https://golfflipper.com/best-hybrid-golf-clubs-of-all-time/), double-check your command for accuracy, ensure the constructor arguments are properly formatted in NFT_var.ts, and make sure you're using the right Hardhat version for compatibility (like 2.8.12). Verify your contract again with accurate inputs, and the issue should be resolved.

**Comment by Jaber-Saed** [comment]
The error you're encountering while trying to verify your NFT smart contract on the Apothem Network using Hardhat's verify command seems to be related to a network response that is not in the expected JSON format. This error often occurs when the response from the verification service isn't what the command is expecting.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Retry Verification:**
   - Sometimes, network issues or temporary glitches can cause unexpected responses. Try running the verification command again.

2. **Check Network Connection:**
   - Ensure that your network connection is stable and that you have proper internet access.

3. **Check Verification Service Status:**
   - The error might be related to the verification service itself. Check if the verification service that Hardhat uses (e.g., Etherscan, Polygonscan) is experiencing any issues. It's possible that the service is temporarily unavailable or experiencing problems.

4. **Update Hardhat:**
   - Make sure you're using the latest version of Hardhat and the plugins related to verification. You mentioned you're using Hardhat 0.8.12, but newer versions might have improvements or bug fixes.

5. **Verify Manually:**
   - If the automatic verification process continues to fail, you can consider verifying the contract manually on the verification service's website. This might require more steps and the copying of certain contract information to the service's verification form.

6. **Network Configuration:**
   - Double-check your Hardhat configuration for the Apothem Network. Make sure all the necessary information, such as the RPC URL and contract address, is correctly specified in your Hardhat configuration file.

7. **Error Handling:**
   - If the error message you provided is coming from the Undici library (which is used for making HTTP requests), there might be an issue with the HTTP response. It's possible that the response you're receiving is not in the expected format.

8. **Documentation and Community:**
   - Check the official Hardhat documentation, GitHub repository, or community forums for any known issues or solutions related to verifying contracts on the Apothem Network.

9. **Contact Support:**
   - If none of the above steps work, consider reaching out to the support channels of the specific network or Hardhat for assistance.

Remember that verifying smart contracts can sometimes be a finicky process due to the complexities involved in interacting with external verification services. If the issue persists, it's a good idea to seek help from the community or the support channels of the tools you're using.
